### ISSUE TITLE
updating one backend invalidates all backends

### ISSUE THREAD

**Issue by sbfaulkner** [fastly/terraform-provider-fastly#198]
NOTE: this is possibly/likely a duplicate of #179 

with initial `fastly.tf`...

```
resource "fastly_service_v1" "tf-test" {
  name = "tf-test"

  domain {
    name = "tf-test.example.com"
  }

  backend {
    name             = "backend 0"
    address          = "tf-test-0.example.com"
  }

  backend {
    name             = "backend 1"
    address          = "tf-test-1.example.com"
  }
}
```

terraform apply result is...

```
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # fastly_service_v1.tf-test will be created
  + resource "fastly_service_v1" "tf-test" {
      + activate       = true
      + active_version = (known after apply)
      + cloned_version = (known after apply)
      + comment        = "Managed by Terraform"
      + default_host   = (known after apply)
      + default_ttl    = 3600
      + id             = (known after apply)
      + name           = "tf-test"

      + backend {
          + address               = "tf-test-0.example.com"
          + auto_loadbalance      = true
          + between_bytes_timeout = 10000
          + connect_timeout       = 1000
          + error_threshold       = 0
          + first_byte_timeout    = 15000
          + max_conn              = 200
          + name                  = "backend 0"
          + port                  = 80
          + ssl_check_cert        = true
          + use_ssl               = false
          + weight                = 100
        }
      + backend {
          + address               = "tf-test-1.example.com"
          + auto_loadbalance      = true
          + between_bytes_timeout = 10000
          + connect_timeout       = 1000
          + error_threshold       = 0
          + first_byte_timeout    = 15000
          + max_conn              = 200
          + name                  = "backend 1"
          + port                  = 80
          + ssl_check_cert        = true
          + use_ssl               = false
          + weight                = 100
        }

      + domain {
          + name = "tf-test.example.com"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fastly_service_v1.tf-test: Creating...
fastly_service_v1.tf-test: Still creating... [10s elapsed]
fastly_service_v1.tf-test: Creation complete after 15s [id=XXXXXXXXXXXXXXXX]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

editing the configuration of any one backend...

```
git diff
diff --git a/terraform/cdn/tf-test/fastly.tf b/terraform/cdn/tf-test/fastly.tf
index b510f119..a0ca0f39 100644
--- a/terraform/cdn/tf-test/fastly.tf
+++ b/terraform/cdn/tf-test/fastly.tf
@@ -8,6 +8,7 @@ resource "fastly_service_v1" "tf-test" {
   backend {
     name               = "backend 0"
     address            = "tf-test-0.example.com"
+    first_byte_timeout = 30000
   }

   backend {
```

terraform plan shows ALL backends being deleted and re-added...

```
terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

fastly_service_v1.tf-test: Refreshing state... [id=XXXXXXXXXXXXXXXX]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # fastly_service_v1.tf-test will be updated in-place
  ~ resource "fastly_service_v1" "tf-test" {
        activate       = true
        active_version = 1
        comment        = "Managed by Terraform"
        default_ttl    = 3600
        id             = "XXXXXXXXXXXXXXXX"
        name           = "tf-test"

      - backend {
          - address               = "tf-test-0.example.com" -> null
          - auto_loadbalance      = true -> null
          - between_bytes_timeout = 10000 -> null
          - connect_timeout       = 1000 -> null
          - error_threshold       = 0 -> null
          - first_byte_timeout    = 15000 -> null
          - max_conn              = 200 -> null
          - name                  = "backend 0" -> null
          - port                  = 80 -> null
          - ssl_check_cert        = true -> null
          - use_ssl               = false -> null
          - weight                = 100 -> null
        }
      + backend {
          + address               = "tf-test-0.example.com"
          + auto_loadbalance      = true
          + between_bytes_timeout = 10000
          + connect_timeout       = 1000
          + error_threshold       = 0
          + first_byte_timeout    = 30000
          + max_conn              = 200
          + name                  = "backend 0"
          + port                  = 80
          + ssl_check_cert        = true
          + use_ssl               = false
          + weight                = 100
        }
      - backend {
          - address               = "tf-test-1.example.com" -> null
          - auto_loadbalance      = true -> null
          - between_bytes_timeout = 10000 -> null
          - connect_timeout       = 1000 -> null
          - error_threshold       = 0 -> null
          - first_byte_timeout    = 15000 -> null
          - max_conn              = 200 -> null
          - name                  = "backend 1" -> null
          - port                  = 80 -> null
          - ssl_check_cert        = true -> null
          - use_ssl               = false -> null
          - weight                = 100 -> null
        }
      + backend {
          + address               = "tf-test-1.example.com"
          + auto_loadbalance      = true
          + between_bytes_timeout = 10000
          + connect_timeout       = 1000
          + error_threshold       = 0
          + first_byte_timeout    = 15000
          + max_conn              = 200
          + name                  = "backend 1"
          + port                  = 80
          + ssl_check_cert        = true
          + use_ssl               = false
          + weight                = 100
        }

        domain {
            name = "tf-test.example.com"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```

**Comment by sbfaulkner** [comment]
further investigation shows that the actual change applied is the single affected backend (ie. this may only be an issue with the reported plan diff)

**Comment by tibbon** [comment]
I'm seeing that even without a change at all (I can apply and then re-plan) it shows it wants to create/destroy the backends.

**Comment by rcollier-goodrx** [comment]
Any chance this good be addressed in a future update? This makes it really hard to review a plan prior to the apply because it shows backends being removed and added.

**Comment by phamann** [comment]
Hi all, just a quick update to say this has now been fixed (#304) and released in version [`v0.20.1`](https://registry.terraform.io/providers/fastly/fastly/0.20.1) of the provider. So am going to close out this issue. Feel free to reopen if you have further issues.
